autoload/ale/fix.vim

```diff
@@ -4,40 +4,15 @@ call ale#Set('fix_on_save_ignore', {})
 " Vim doesn't let you modify hidden buffers.
 function! ale#fix#ApplyQueuedFixes(buffer) abort
     let l:data = get(g:ale_fix_buffer_data, a:buffer, {'done': 0})
-    let l:has_bufline_api = exists('*deletebufline') && exists('*setbufline')
 
-    if !l:data.done || (!l:has_bufline_api && a:buffer isnot bufnr(''))
+    if !l:data.done
```

Comment:
This change will break how fixing files works, because we need to avoid the `remove()` call below to keep the data until the buffer comes back into focus again, because older versions of Vim without the `bufline` API don't let you modify buffers other than the current buffer. You'll need to keep the check here still.

Reply:
The check is still there, i just moved it to the `ale#fix#SetBufferContents`.

Reply:
I thought that the `remove()` call was somewhere else. I've added that check back!